runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/resources/PipelineResourcesTest.java

```diff
@@ -102,8 +100,57 @@ public void testPackagingDirectoryResourceToJarFile() throws IOException {
 
   @Test
   public void testIfThrowsWhenThereIsNoTemporaryFolderForJars() throws IOException {
-    List<String> filesToStage = new ArrayList<>();
-    filesToStage.add(tmpFolder.newFolder().getAbsolutePath());
+    List<String> filesToStage = Arrays.asList(tmpFolder.newFolder().getAbsolutePath());
+
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> PipelineResources.prepareFilesForStaging(filesToStage, null));
+
+    assertEquals(
+        "Please provide temporary location for storing the jar files.", exception.getMessage());
+  }
+
+  @Test
+  public void testPrepareFilesForStagingFromOptions() throws IOException {
+    String nonexistentFilePath = tmpFolder.getRoot().getPath() + "/nonexistent/file";
+    String existingFilePath = tmpFolder.newFile("existingFile").getAbsolutePath();
+    List<String> filesToStage = Arrays.asList(nonexistentFilePath, existingFilePath);
+    String temporaryLocation = tmpFolder.newFolder().getAbsolutePath();
+
+    FileStagingOptions options = PipelineOptionsFactory.create().as(FileStagingOptions.class);
+    options.setFilesToStage(filesToStage);
+    options.setTempLocation(temporaryLocation);
+
+    assertThrows(
+        "To-be-staged file does not exist: ",
+        IllegalStateException.class,
+        () -> PipelineResources.prepareFilesForStaging(filesToStage, temporaryLocation));
+  }
+
+  @Test
+  public void testPackagingDirectoryResourceFromOptions() throws IOException {
+    String directoryPath = tmpFolder.newFolder().getAbsolutePath();
+    List<String> filesToStage = Arrays.asList(directoryPath);
+    String temporaryLocation = tmpFolder.newFolder().getAbsolutePath();
+
+    FileStagingOptions options = PipelineOptionsFactory.create().as(FileStagingOptions.class);
+    options.setFilesToStage(filesToStage);
+    options.setTempLocation(temporaryLocation);
+
+    PipelineResources.prepareFilesForStaging(options);
+    List<String> result = options.getFilesToStage();
+
+    assertTrue(new File(result.get(0)).exists());
```

Comment:
Should we also assert there is exactly one item in result?

Reply:
yep, done